tdrs-backend/tdpservice/scheduling/tasks.py

```diff
@@ -0,0 +1,112 @@
+"""schedule tasks."""
+from __future__ import absolute_import
+# The tasks
+
+import hashlib
+import os
+
+from celery import shared_task
+from django.conf import settings
+import datetime
+import paramiko
+import logging
+from tdpservice.data_files.models import DataFile, LegacyFileTransfer
+
+logger = logging.getLogger(__name__)
+
+
+def write_key_to_file(private_key):
```

Comment:
Should this function be nested in `upload()` just like `create_dir()` ?

Reply:
I agree with this, it's always more readable to have a nested associated method